tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -878,9 +997,10 @@ TEST_CASE("Test Tap position optimizer") {
             SUBCASE("voltage band") {
                 state_b.rank = 0;
                 state_b.u_pu = [&state_b, &regulator_b](ControlSide side) {
-                    CHECK(side == regulator_b.control_side());
-
-                    // tap pos closer to tap_max at tap side <=> lower voltage at control side
+                    if (state_b.tap_side == regulator_b.control_side()) {
+                        return static_cast<DoubleComplex>(
+                            test::normalized_lerp(state_b.tap_pos, state_b.tap_min, state_b.tap_max));
+                    }
                     return static_cast<DoubleComplex>(
                         test::normalized_lerp(state_b.tap_pos, state_b.tap_max, state_b.tap_min));
```

Comment:
since this code repeats fairly often, maybe you can move it outside this function?

I also see that you use
```cpp
                    bool const control_at_tap = side == regulator_b.control_side();
                    auto const value = test::normalized_lerp(state_b.tap_pos, state_b.tap_min, state_b.tap_max);

                    auto const value = control_at_tap
                                           ? test::normalized_lerp(state_b.tap_pos, state_b.tap_max, state_b.tap_min)
                                           : test::normalized_lerp(state_b.tap_pos, state_b.tap_min, state_b.tap_max);
                    return DoubleComplex{value, value};
```
somewhere else, which kind of amounts to the same thing. That can be homogenized to reduce cognitive complexity.

Reply:
feel free to resolve this. I think it looks good as is. More is always possible, but probably not necessary so I leave that up to you.